context/app/api/client.py

```diff
@@ -167,6 +168,55 @@ def _get_mock_vitessce_conf(self):
             ]
         }
 
+    def get_most_recent_dataset_for_assay(self, assay):
+        query = {
+            "size": 1,
+            "query": {
+                "bool": {
+                    "must_not": {
+                        "exists": {
+                            "field": "next_revision_uuid"
+                        }
+                    }
+                }
+            },
+            "post_filter": {
+                "bool": {
+                    "must": [
+                        {
+                            "term": {
+                                "data_types.keyword": assay
+                            }
+                        },
+                        {
+                            "term": {
+                                "entity_type.keyword": "Dataset"
+                            }
+                        }
+                    ]
+                }
+            },
+            "sort": [{"mapped_last_modified_timestamp.keyword": "desc"}]
+        }
+
+        response_json = self._request(
+            current_app.config['ELASTICSEARCH_ENDPOINT']
+            + current_app.config['PORTAL_INDEX_PATH'],
+            body_json=query)
+        return response_json['hits']['hits'][0]["_source"] if len(response_json['hits']['hits']) > 0 else None
+
+    def get_assays_with_visualizations(self):
+        tc = TypeClient(current_app.config["TYPE_SERVICE_ENDPOINT"])
```

Comment:
The goal is a script a developer can run to produce an up-to-date list of visualize-able dataset types? I would put it in a separate directory from the runtime code, and either hardcode the type service endpoint, or make it a command-line parameter.

(One possible shape for this: The output is checked in as a configuration file, and the test suite checks that there aren't new visualizable assay types.) (Having dependencies on live APIs during tests isn't great, but this would be small, limited scope... and it's really a consequence of using microservices for things that might have been better as shared configuration. But that's not worth fighting at this point.)